HW_1.py

```diff
@@ -0,0 +1,440 @@
+    
+    def show_chat(self):
+        for message in self.chat_history[::-1]:
+            message.show()
+            print()
+            
+    def recieve(self, message):
+        self.chat_history = [message] + self.chat_history
+
+class Message:
+    # Ваш код здесь
+    def __init__(self, text, user):
+        self.text = text
+        self.datetime = datetime
+        self.user = user
+        
+    def show(self):
+        print(self.user.show_name(), str(self.datetime), '\n', self.text, '\n________________\n')
+    
+    def send(self, chat):
+        self.datetime = datetime.now()
+        chat.recieve(self)
+
+class User:
+    # Ваш код здесь
+    def __init__(self, nickname, greeting):
+        self.nickname = nickname
+        self.greeting = greeting
+    def show_name(self):
+        return self.nickname
+
+
+# In[292]:
+
+
+chatik = Chat()
+
+
+# In[283]:
+
+
+Poirot = User('Poirot', 'Bonjour')
+Hastings = User('Hastings', 'Good day!')
+
+
+# In[293]:
+
+
+Poirot_message = Message('There are many things not called poison which can kill a man', Poirot).send(chatik)
+Hastings_message = Message('Jesus, Mary and Joseph and the wee donkey!', Hastings).send(chatik)
+
+
+# In[294]:
+
+
+chatik.show_chat()
+
+
+# # Задание 2 (3 балла)
+
+# В питоне как-то слишком типично и неинтересно происходят вызовы функций. Напишите класс `Args`, который будет хранить в себе аргументы, а функции можно будет вызывать при помощи следующего синтаксиса.
+# 
+# Использовать любые модули **нельзя**, да и вряд-ли это как-то поможет)
+
+# In[88]:
+
+
+class Args:
+    def __init__(self, *args, **kwargs):
+        self.args = args
+        self.kwargs = kwargs
+    
+    def __rlshift__(self, other):
+        return other(*self.args, **self.kwargs)
+
+
+# In[89]:
+
+
+sum << Args([1, 2])
+
+
+# In[90]:
+
+
+(lambda a, b, c: a**2 + b + c) << Args(1, 2, c=50)
+
+
+# # Задание 3 (5 баллов)
+
+# Сделайте класс наследник `float`. Он должен вести себя как `float`, но также должен обладать некоторыми особенностями:
+# + При получении атрибутов формата `<действие>_<число>` мы получаем результат такого действия над нашим числом
+# + Создавать данные атрибуты в явном виде, очевидно, не стоит
+# 
+# Подсказка: если в процессе гуглёжки, вы выйдете на такую тему как **"Дескрипторы", то это НЕ то, что вам сейчас нужно**
+# 
+# Примеры использования ниже
+
+# In[163]:
+
+
+class StrangeFloat(float):
+    def __getattribute__(self, item):
+        if '__' not in item:
+            operation, value = item.split('_')
+            if operation == 'add':
+                    return type(self)(self + float(value))
+            elif operation == 'subtract':
+                    return type(self)(self - float(value))
+            elif operation == 'multiply':
+                    return type(self)(self * float(value))
+            elif operation == 'divide':
+                    return type(self)(self / float(value))
+            else:
+                raise AttributeError
```

Comment:
Не работает последний пример с встроенным методом `float`а. Чинится это вот так
```suggestion
class StrangeFloat(float):
    def __getattribute__(self, item):
        if item not in dir(float):
            operation, value = item.split('_')
            if operation == 'add':
                    return type(self)(self + float(value))
            elif operation == 'subtract':
                    return type(self)(self - float(value))
            elif operation == 'multiply':
                    return type(self)(self * float(value))
            elif operation == 'divide':
                    return type(self)(self / float(value))
            else:
                raise AttributeError
        else:
            return super().__getattribute__(item)
```